tests/unit/cf/test_cf_tasks.py

```diff
@@ -0,0 +1,87 @@
+from unittest.mock import patch
+from harvester.utils import CFHandler
+import collections
+
+
+class AppMock:
+    def __init__(self, x):
+        self.tasks = self.tasksMock
+        self.recent_logs = self.logsMock
+
+    def tasksMock(x):
+        return [1, 2, 3]
+
+    def logsMock(x):
+        return [{"cf_task_integration": 1}, 2, 3]
+
+    def __getitem__(self, x):
+        return self
+
+
+class V2Mock:
+    def __init__(self, x):
+        self.apps = AppMock(x)
+
+
+class V3Mock:
+    def __init__(self, x):
+        self.apps = AppMock(x)
+
+
+class CFClientMock:
+    def __init__(self, x):
+        self.v2 = V2Mock(x)
+        self.v3 = V3Mock(x)
+        pass
+
+    def init_with_user_credentials(self, a, b):
+        pass
+
+
+class TaskManagerMock:
+    def __init__(self, x, y):
+        pass
+
+    def create(self, a, b, c):
+        return collections.defaultdict(list)
+
+    def get(self, a):
+        return collections.defaultdict(list)
+
+    def cancel(self, a):
+        return "cancelled"
+
+
+with patch("harvester.utils.CloudFoundryClient", CFClientMock), patch(
+    "harvester.utils.TaskManager", TaskManagerMock
+):
+    CFUtil = CFHandler("url", "user", "password")
+
+
```

Comment:
could we just patch the CFHandler class instead of all these underlying classes in the libraries?

Reply:
We could, but then we're just testing our mocks. By only patching the supporting libraries we aren't responsible for, we ensure that the CFHandler logic is sound. 

The mock objects could probably be improved though.

Reply:
okay yeah i see what you're doung